File: testdata/binding_gap/python/dup_methods.py
# 두 클래스가 같은 이름의 메서드를 가진 경우 nameIndex dedup 오병합 측정용 fixture.
# - Alpha.save: @classmethod 데코레이터 1개 (StartLine은 데코레이터 줄)
# - Beta.save: @classmethod 데코레이터 1개 (StartLine은 데코레이터 줄, Alpha.save보다 아래)
# 동명 메서드가 서로 다른 클래스 본문에 존재하지만 nameIndex["save"]는 하나만 보관한다.


class Alpha:
    @classmethod
    def save(cls) -> int:
        """@intent Alpha save"""
        return 1


class Beta:
    @classmethod
    def save(cls) -> int:
        """@intent Beta save"""
        return 2

File: testdata/binding_gap/typescript/dup_methods.ts
// 두 클래스가 같은 이름의 메서드를 가진 경우 nameIndex dedup 오병합 측정용 fixture.
// - Alpha.render: @Memoized 데코레이터 있음
// - Beta.render: @Memoized 데코레이터 있음 (Alpha.render보다 아래)

declare function Memoized(target: any, key: string): void;

export class Alpha {
    @Memoized
    render(): string {
        return "alpha";
    }
}

export class Beta {
    @Memoized
    render(): string {
        return "beta";
    }
}
